emission/individual_tests/TestNominatim.py

```diff
@@ -0,0 +1,113 @@
+        for k in key_list:
+            self.assertEqual(nominatim_result[k], geofabrik_result[k])
+
+    #Checks the display name generated by get_filtered_place in clean_and_resample.py, which creates a cleaned place from the fake place
+    # and reverse geocodes with the coordinates.
+    def test_get_filtered_place(self):
+        raw_result = ecww.WrapperBase.__getattr__(clean.get_filtered_place(fake_place), "data")
+        print(NOMINATIM_QUERY_URL)
+        actual_result = ecww.WrapperBase.__getattr__(raw_result, "display_name")
+        expected_result = "Dorrance Street, Providence"
+        self.assertEqual(expected_result, actual_result)
+
+    #Testing make_url_geo, which creates a query URL from the input string. 
+    def test_make_url_geo(self):
+        expected_result = NOMINATIM_QUERY_URL + "/search?q=Providence%2C+Rhode+Island&format=json"
+        actual_result = eco.Geocoder.make_url_geo("Providence, Rhode Island")
+        self.assertEqual(expected_result, actual_result)
+
+    #Testing get_json_geo, which passes in an address as a query. Compares three select k,v pairs in the results. 
+    def test_get_json_geo(self):
+        expected_result = {'place_id': 132490, 'licence': 'Data © OpenStreetMap contributors, ODbL 1.0. https://osm.org/copyright', 'osm_type': 'way', 'osm_id': 141567710, 'boundingbox': ['41.8325787', '41.8332278', '-71.4161848', '-71.4152064'], 'lat': '41.8330097', 'lon': '-71.41568124868104', 'display_name': 'State of Rhode Island Department of Administration, 1, Park Street, Downtown, Providence, Providence County, 02908, United States', 'class': 'building', 'type': 'civic', 'importance': 1.75001}
+        actual_result = eco.Geocoder.get_json_geo("State of Rhode Island Department of Administration, 1, Park Street, Downtown, Providence, Providence County, 02908, United States")[0]
+        key_list = ['osm_id', 'boundingbox', 'display_name']
+        for k in key_list:
+            self.assertEqual(expected_result[k], actual_result[k])
+
+    #Testing the geocode function, which passes in an address and gets latitude and longitude.
+    # Test creates instance of coordinates using coordinate class. Getting lat and lon of the coordinate using get_lat and get_lon methods from the class.
+    def test_geocode(self):
+        expected_result_lon = Coordinate(41.8239891, -71.4128343).get_lon()
+        expected_result_lat = Coordinate(41.8239891, -71.4128343).get_lat()
+        actual_result = eco.Geocoder.geocode("Providence, Rhode Island")
+        actual_result_lon = actual_result.get_lon()
+        actual_result_lat = actual_result.get_lat()
+        self.assertEqual(expected_result_lon, actual_result_lon)
+        self.assertEqual(expected_result_lat, actual_result_lat)
+
+    #Testing make_url_reverse, which creates a query url from a lat and lon
+    def test_make_url_reverse(self):
+        expected_result = NOMINATIM_QUERY_URL + "/reverse?lat=41.8239891&lon=-71.4128343&format=json"
```

Comment:
why is this not using `make_url_geo`?

Reply:
I made a separate test for each method; makre_url_geo passes in an address, and make_url_reverse passes in lat/lon. Do you think this test is redundant+should be removed?

Reply:
I don't remember what I meant by this comment any more, reverting..